packages/core-v2/src/semantic-model/aggregator/aggregator.ts

```diff
@@ -80,20 +95,45 @@ class SemanticModelAggregatorInternal implements SemanticModelAggregator {
     private notifyBaseModel(fromModel: SupportedModels, updated: AggregatedEntityWrapper[], removed: string[]) {
         // This does only a simple merge
         this.baseModelEntities = this.baseModelEntities
-            .filter(entity => !removed.includes(entity.id) && !updated.find(updatedEntity => updatedEntity.id === entity.id))
+            .filter(
+                (entity) =>
+                    !removed.includes(entity.id) && !updated.find((updatedEntity) => updatedEntity.id === entity.id)
+            )
             .concat(updated);
-        this.baseModelSubscribers.forEach(subscriber => subscriber(updated, removed));
+        this.baseModelSubscribers.forEach((subscriber) => subscriber(updated, removed));
     }
 
     getView(): SemanticModelAggregatorView {
         return new SemanticModelAggregatorView(this);
     }
+
+    addVisualModel(model: VisualEntityModel) {
```

Comment:
Viz první koment. Tady mi přijde, že kopíruješ logiku `addModel`u, ale do hloubky jsem to nezkoumal, takže nechám na tobě.